reflex/app.py

```diff
@@ -420,25 +423,23 @@ def add_custom_404_page(
             title: The title of the page.
             description: The description of the page.
             image: The image to display on the page.
+            on_load: The event handler(s) that will be called each time the page load.
             meta: The metadata of the page.
         """
-        title = title or constants.TITLE_404
-        image = image or constants.FAVICON_404
-        description = description or constants.DESCRIPTION_404
-
-        component = component if isinstance(component, Component) else component()
-
-        compiler_utils.add_meta(
-            component,
-            title=title,
-            image=image,
-            description=description,
+        self.add_page(
+            component=(
+                component
+                if isinstance(component, Component)
+                else (component() if component else Fragment.create())
```

Comment:
This check is already done in `add_page`, so we don't need to do it here actually.

Reply:
We need to for the case where component is None, maybe I can simplify it here.